addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -83,7 +84,8 @@ options.registry.TableOfContent = options.Class.extend({
         // We dispose the scrollSpy because the navbar will be updated.
         wUtils.disposeScrollSpy(this);
         const $nav = this.$target.find('.s_table_of_content_navbar');
-        const $headings = this.$target.find(this.targetedElements);
+        const $headings = this.$target.find(this.targetedElements).toArray()
+            .filter(el => !el.closest('.o_snippet_desktop_invisible'));
```

Comment:
I did not tested but here you are in options.js, meaning the navbar of table of content is generated during edit mode... are you sure this makes sense and work outside of edit mode as a visitor? = the navbar displays the section which are currently visible on the current device's size? I don't understand how that can work without replicating the d-none/d-md-xxx classes on the navbar links.

Reply:
The nav bar of the table of content is not displayed in mobile view so we can ignore the headings visible only in mobile view when we build the nav. I add a comment to explain that :+1: